### ISSUE TITLE
Android Build Fails on 0.5

### ISSUE THREAD

**Issue by sotrh** [gfx-rs/wgpu#699]
I was trying to get my [pipeline tutorial code](https://sotrh.github.io/learn-wgpu/beginner/tutorial3-pipeline/#what-s-a-pipeline) working on Android with [cargo-apk](https://crates.io/crates/cargo-apk). The build succeeds until it gets to wgpu-native (v0.5.0)

```bash
   Compiling wgpu-native v0.5.0
error[E0432]: unresolved import `raw_window_handle::unix`
  --> /home/benjamin/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-native-0.5.0/src/device.rs:91:28
   |
91 |     use raw_window_handle::unix::XlibHandle;
   |                            ^^^^ could not find `unix` in `raw_window_handle`

error[E0432]: unresolved import `raw_window_handle::unix`
   --> /home/benjamin/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-native-0.5.0/src/device.rs:105:28
    |
105 |     use raw_window_handle::unix::WaylandHandle;
    |                            ^^^^ could not find `unix` in `raw_window_handle`

error[E0599]: no variant or associated item named `Xlib` found for enum `raw_window_handle::RawWindowHandle` in the current scope
  --> /home/benjamin/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-native-0.5.0/src/device.rs:51:14
   |
51 |         Rwh::Xlib(h) => core::instance::Surface {
   |              ^^^^ variant or associated item not found in `raw_window_handle::RawWindowHandle`

error[E0599]: no variant or associated item named `Wayland` found for enum `raw_window_handle::RawWindowHandle` in the current scope
  --> /home/benjamin/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-native-0.5.0/src/device.rs:58:14
   |
58 |         Rwh::Wayland(h) => core::instance::Surface {
   |              ^^^^^^^ variant or associated item not found in `raw_window_handle::RawWindowHandle`

error[E0599]: no method named `create_surface_from_xlib` found for reference `&gfx_backend_vulkan::Instance` in the current scope
  --> /home/benjamin/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-native-0.5.0/src/device.rs:55:34
   |
55 |                 .map(|inst| inst.create_surface_from_xlib(h.display as _, h.window as _)),
   |                                  ^^^^^^^^^^^^^^^^^^^^^^^^ help: there is a method with a similar name: `create_surface_android`

error[E0599]: no method named `create_surface_from_wayland` found for reference `&gfx_backend_vulkan::Instance` in the current scope
  --> /home/benjamin/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-native-0.5.0/src/device.rs:62:34
   |
62 |                 .map(|inst| inst.create_surface_from_wayland(h.display, h.surface)),
   |                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `&gfx_backend_vulkan::Instance`

error[E0599]: no variant or associated item named `Xlib` found for enum `raw_window_handle::RawWindowHandle` in the current scope
  --> /home/benjamin/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-native-0.5.0/src/device.rs:92:61
   |
92 |     wgpu_create_surface(raw_window_handle::RawWindowHandle::Xlib(XlibHandle {
   |                                                             ^^^^ variant or associated item not found in `raw_window_handle::RawWindowHandle`

error[E0599]: no variant or associated item named `Wayland` found for enum `raw_window_handle::RawWindowHandle` in the current scope
   --> /home/benjamin/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-native-0.5.0/src/device.rs:106:61
    |
106 |     wgpu_create_surface(raw_window_handle::RawWindowHandle::Wayland(WaylandHandle {
    |                                                             ^^^^^^^ variant or associated item not found in `raw_window_handle::RawWindowHandle`

error: aborting due to 8 previous errors

Some errors have detailed explanations: E0432, E0599.
For more information about an error, try `rustc --explain E0432`.
error: could not compile `wgpu-native`.

To learn more, run the command again with --verbose.
Error: Command 'cargo build --target armv7-linux-androideabi' had a non-zero exit code.
```

**Repro steps**
1. Clone the android branch of sotrh/learn-wgpu https://github.com/sotrh/learn-wgpu/tree/android
2. Copy the `code/showcase/android/` directory to it's own directory outside the project. (cargo-apk seems to have trouble with workspace projects)
3. Install cargo-apk
```bash
cargo install cargo-apk
```
4. In the copied `android` directory run
```bash
cargo apk run
```

The issue seems to stem from src/device.rs.

**Platform**

Linux Mint 19
wgpu v0.5.0

**Comment by kvark** [comment]
Thank you for filing!
Looks like 0.5 is indeed broken for Android.
This should work in `master`, we even have a [ci job](https://github.com/gfx-rs/wgpu/blob/277cf8d3bf4958f5278482049639cbec62f274ab/.github/workflows/ci.yml#L21) for it.
Leaving the issue open to make a patch for 0.5 version, should be easy!

**Comment by PrototypeNM1** [comment]
Looks like this issue can be closed now that 0.6 has released.
